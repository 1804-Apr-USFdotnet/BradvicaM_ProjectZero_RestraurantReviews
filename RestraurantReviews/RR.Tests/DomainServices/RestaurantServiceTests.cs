using ApprovalTests;
using ApprovalTests.Reporters;
using Autofac;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using Moq;
using RR.Console;
using RR.DomainContracts;
using RR.DomainServices;
using RR.Models;
using RR.RepositoryContracts;

namespace RR.Tests.DomainServices
{
    [TestClass]
    public class RestaurantServiceTests
    {
        private readonly Mock<IRestaurantRepository> _mock;

        public RestaurantServiceTests()
        {
            _mock = new Mock<IRestaurantRepository>();
            _mock.Setup(x => x.Add(It.IsAny<Restaurant>()));
        }

        [TestMethod]
        [UseReporter(typeof(DiffReporter))]
        public void TopThreeRestaurantByAverageReview_Returns_CorrectRestaurants()
        {
            using (var container = Bootstrapper.RegisterTypes())
            {
                var service = container.Resolve<IRestaurantService>();

                var results = service.TopThreeRestaurantByAverageReview();

                Approvals.VerifyAll(results, "Restaurants");
            }
        }

        [TestMethod]
        [UseReporter(typeof(DiffReporter))]
        public void AllRestaurants_Returns_AllRestaurantsFromDatabase()
        {
            using (var container = Bootstrapper.RegisterTypes())
            {
                var service = container.Resolve<IRestaurantService>();

                var results = service.AllRestaurants();

                Approvals.VerifyAll(results, "Restaurants");
            }
        }

        [TestMethod]
        [UseReporter(typeof(DiffReporter))]
        public void SearchByString_Returns_AllRestaurantsThatContainStringValue()
        {
            using (var container = Bootstrapper.RegisterTypes())
            {
                var service = container.Resolve<IRestaurantService>();

                var results = service.SearchAll("Jordan");

                Approvals.VerifyAll(results, "Restaurants");
            }
        }

        [TestMethod]
        public void AddRestaurant_Correctly_AddsRestuarantToDatabase()
        {
            var service = new RestaurantService(_mock.Object);

            var resturant = new Restaurant();

            service.AddRestaurant(resturant);

            _mock.Verify(x => x.Add(It.IsAny<Restaurant>()), Times.AtLeastOnce);
        }

        [TestMethod]
        [UseReporter(typeof(DiffReporter))]
        public void SearchByName_GivenString_ReturnsCorrectRestaurant()
        {
            using (var container = Bootstrapper.RegisterTypes())
            {
                var service = container.Resolve<IRestaurantService>();

                var result = service.SearchByName("Billy Bobs Texas BBQ");

                Approvals.Verify(result);
            }
        }
    }
}
