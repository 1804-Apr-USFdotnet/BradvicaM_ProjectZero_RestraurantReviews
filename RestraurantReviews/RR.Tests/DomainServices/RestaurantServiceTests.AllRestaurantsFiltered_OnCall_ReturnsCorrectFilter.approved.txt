Resaurants[0] = 
Id: 2d7bfe1e-04cb-4d5d-b93a-7aa673eedeeb
Name: Babbleset 
Street: 4489 Sugar Plaza
City: Terre Haute
State: IN
ZipCode: 47812
PhoneNumber: 812 386 4708
Website: tinypic.com
Rating: 7

Resaurants[1] = 
Id: 556243ce-2eff-4d60-b991-5a122d504d86
Name: Babblestorm 
Street: 4532 Golden Leaf Circle
City: Eugene
State: OR
ZipCode: 97405
PhoneNumber: 541 991 6377
Website: reference.com
Rating: 3

Resaurants[2] = 
Id: 5e90e5d5-0cc9-4ce1-9526-6906e9a4a41c
Name: Babblestorm 
Street: 8351 Parkside Pass
City: Baltimore
State: MD
ZipCode: 21203
PhoneNumber: 410 689 7772
Website: flavors.me
Rating: 1

Resaurants[3] = 
Id: 2087ed3f-f140-4ffb-937e-6588a401a2a2
Name: Billy Bobs Texas BBQ 
Street: 123 Barbeque St.
City: Kansas City
State: MO
ZipCode: 81721
PhoneNumber: 8761234121
Website: www.billybobs.com
Rating: 5.88

Resaurants[4] = 
Id: cb76ceb5-3275-4e85-bf54-01059433d564
Name: Dabtype 
Street: 54894 Fisk Avenue
City: Cleveland
State: OH
ZipCode: 44191
PhoneNumber: 216 860 1827
Website: t.co
Rating: 8

Resaurants[5] = 
Id: 6136953b-4ce3-4b12-b5ca-6d5dca4b5b6e
Name: Dabvine 
Street: 1166 Onsgard Point
City: Los Angeles
State: CA
ZipCode: 90005
PhoneNumber: 310 878 5505
Website: jugem.jp
Rating: 4

Resaurants[6] = 
Id: 613fb79f-3bd9-4145-9718-39ea8d65abfc
Name: Divape 
Street: 62616 Vernon Terrace
City: Largo
State: FL
ZipCode: 33777
PhoneNumber: 727 757 3243
Website: prweb.com
Rating: 5

Resaurants[7] = 
Id: 46537979-c02f-4b0c-a168-2830a8be8a6a
Name: Feedfire 
Street: 415 Scott Terrace
City: Aiken
State: SC
ZipCode: 29805
PhoneNumber: 803 317 8206
Website: twitter.com
Rating: 5

Resaurants[8] = 
Id: 25e1611e-b5ae-4f4f-aa0e-2e8d341b71b9
Name: Jatri 
Street: 99 Warbler Alley
City: Boulder
State: CO
ZipCode: 80328
PhoneNumber: 303 998 3393
Website: blogspot.com
Rating: 2

Resaurants[9] = 
Id: 891b63c8-b053-4cfe-8310-55f6cf7ffc56
Name: Jaxbean 
Street: 4 Jenna Parkway
City: Van Nuys
State: CA
ZipCode: 91406
PhoneNumber: 323 874 6200
Website: umich.edu
Rating: 7

Resaurants[10] = 
Id: c6a14303-ea6c-4249-a4ab-3c931a5755ce
Name: Kare 
Street: 73444 Almo Plaza
City: Saint Petersburg
State: FL
ZipCode: 33737
PhoneNumber: 727 273 3457
Website: yelp.com
Rating: 1

Resaurants[11] = 
Id: ec5b9d05-9132-45f6-bd05-58766818c6c2
Name: Kimia 
Street: 684 Hayes Trail
City: Los Angeles
State: CA
ZipCode: 90189
PhoneNumber: 213 557 7358
Website: economist.com
Rating: 2

Resaurants[12] = 
Id: 59e8782e-a78e-4f77-aed9-407ba0400f3f
Name: Meevee 
Street: 6947 Butternut Plaza
City: Portland
State: OR
ZipCode: 97201
PhoneNumber: 503 602 3108
Website: google.nl
Rating: 7

Resaurants[13] = 
Id: a17388a5-d58a-4a0e-9e7d-3e15c87ce2e0
Name: Minyx 
Street: 271 Mosinee Alley
City: Minneapolis
State: MN
ZipCode: 55470
PhoneNumber: 612 187 8474
Website: bloomberg.com
Rating: 1

Resaurants[14] = 
Id: d373b238-07d4-4733-a1e7-6be6fdb67779
Name: Omba 
Street: 6015 Fallview Park
City: Waco
State: TX
ZipCode: 76705
PhoneNumber: 254 540 4654
Website: hp.com
Rating: 6.195

Resaurants[15] = 
Id: b14b95b7-c0cf-457c-95bb-3a0340989f0c
Name: Riffpath 
Street: 56240 Fisk Street
City: Colorado Springs
State: CO
ZipCode: 80920
PhoneNumber: 719 489 5117
Website: exblog.jp
Rating: 4

Resaurants[16] = 
Id: 18364b3a-dbfe-4510-9125-794f323cefc4
Name: Roomm 
Street: 5 Dwight Point
City: Oklahoma City
State: OK
ZipCode: 73190
PhoneNumber: 405 357 2343
Website: paginegialle.it
Rating: 8

Resaurants[17] = 
Id: 9be1738f-a1b8-4cd0-a033-1c0163a93c87
Name: Shufflester 
Street: 748 Nancy Lane
City: Amarillo
State: TX
ZipCode: 79159
PhoneNumber: 806 714 0740
Website: admin.ch
Rating: 7

Resaurants[18] = 
Id: c325cd17-4619-48bd-97c9-7b2e29db6de9
Name: Topicblab 
Street: 9407 Dorton Terrace
City: Ogden
State: UT
ZipCode: 84403
PhoneNumber: 801 302 4842
Website: auda.org.au
Rating: 10

Resaurants[19] = 
Id: ac93c1fd-4692-4e87-93d2-1ebb122aced3
Name: Topiczoom 
Street: 720 Hudson Park
City: Indianapolis
State: IN
ZipCode: 46226
PhoneNumber: 317 888 5181
Website: yahoo.co.jp
Rating: 5

Resaurants[20] = 
Id: 5d268e7f-900f-45e6-8ee2-34f450c8c77b
Name: Voomm 
Street: 81039 Caliangt Hill
City: Durham
State: NC
ZipCode: 27717
PhoneNumber: 919 968 3437
Website: nba.com
Rating: 2

Resaurants[21] = 
Id: 7e9f9924-156e-4b9f-8a41-0f21137e5df4
Name: Yodel 
Street: 95612 Mayer Park
City: Salt Lake City
State: UT
ZipCode: 84170
PhoneNumber: 801 508 2699
Website: eventbrite.com
Rating: 10

Resaurants[22] = 
Id: ae1d4254-a32b-40c1-ba52-0be618cbe695
Name: Zoombeat 
Street: 253 Sachtjen Park
City: Washington
State: DC
ZipCode: 20244
PhoneNumber: 202 443 8037
Website: samsung.com
Rating: 2

Resaurants[23] = 
Id: 05fc73c3-79f8-4acc-8d2e-0cebbe283c80
Name: Zoovu 
Street: 98 Meadow Ridge Park
City: Sacramento
State: CA
ZipCode: 95823
PhoneNumber: 916 697 7385
Website: chronoengine.com
Rating: 5

